File: backend/api/app/Models/PayrollSlip.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class PayrollSlip extends Model
{
    protected $fillable = [
        'user_id',
        'period_month',
        'nip',
        'name',
        'unit',
        'position',
        'total_income',
        'total_deduction',
        'net_income',
        'sign_location',
        'sign_date',
    ];

    protected $casts = [
        'total_income' => 'decimal:2',
        'total_deduction' => 'decimal:2',
        'net_income' => 'decimal:2',
        'sign_date' => 'date',
    ];

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class);
    }

    public function items(): HasMany
    {
        return $this->hasMany(PayrollSlipItem::class)->orderBy('sort_order');
    }

    public function incomeItems(): HasMany
    {
        return $this->hasMany(PayrollSlipItem::class)
            ->where('type', 'income')
            ->where('amount', '>', 0)
            ->orderBy('sort_order');
    }

    public function deductionItems(): HasMany
    {
        return $this->hasMany(PayrollSlipItem::class)
            ->where('type', 'deduction')
            ->where('amount', '>', 0)
            ->orderBy('sort_order');
    }
}

File: mobile/lib/models/payroll_slip.dart
class PayrollSlipItem {
  const PayrollSlipItem({
    required this.label,
    required this.amount,
  });

  final String label;
  final double amount;

  factory PayrollSlipItem.fromJson(Map<String, dynamic> json) {
    return PayrollSlipItem(
      label: json['label'] as String? ?? '',
      amount: (json['amount'] as num?)?.toDouble() ?? 0,
    );
  }
}

class PayrollSlip {
  const PayrollSlip({
    required this.id,
    required this.periodMonth,
    this.nip,
    this.name,
    this.unit,
    this.position,
    required this.totalIncome,
    required this.totalDeduction,
    required this.netIncome,
    this.signLocation,
    this.signDate,
    this.incomeItems = const [],
    this.deductionItems = const [],
    this.createdAt,
  });

  final int id;
  final String periodMonth;
  final String? nip;
  final String? name;
  final String? unit;
  final String? position;
  final double totalIncome;
  final double totalDeduction;
  final double netIncome;
  final String? signLocation;
  final String? signDate;
  final List<PayrollSlipItem> incomeItems;
  final List<PayrollSlipItem> deductionItems;
  final String? createdAt;

  factory PayrollSlip.fromJson(Map<String, dynamic> json) {
    return PayrollSlip(
      id: json['id'] as int,
      periodMonth: json['period_month'] as String? ?? '',
      nip: json['nip'] as String?,
      name: json['name'] as String?,
      unit: json['unit'] as String?,
      position: json['position'] as String?,
      totalIncome: (json['total_income'] as num?)?.toDouble() ?? 0,
      totalDeduction: (json['total_deduction'] as num?)?.toDouble() ?? 0,
      netIncome: (json['net_income'] as num?)?.toDouble() ?? 0,
      signLocation: json['sign_location'] as String?,
      signDate: json['sign_date'] as String?,
      incomeItems: (json['income_items'] as List<dynamic>?)
              ?.map((e) => PayrollSlipItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      deductionItems: (json['deduction_items'] as List<dynamic>?)
              ?.map((e) => PayrollSlipItem.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
      createdAt: json['created_at'] as String?,
    );
  }
}
